src/javascript/events/component-view.js

```diff
@@ -0,0 +1,58 @@
+/*global module, require */
+'use strict'; // eslint-disable-line strict
+
+const Core = require('../core');
+const getTrace = require('../../libs/get-trace');
+const utils = require('../utils');
+
+const decorateEventData = (eventData, viewedEl, opts) => {
+
+	const { trace, customContext } = getTrace(viewedEl);
+	const context = typeof opts.callback === 'function' ? opts.callback(viewedEl) : {};
+
+	context.domPathTokens = trace;
+	context.url = window.document.location.href || null;
+
+	utils.assignIfUndefined(customContext, context);
+
+	eventData.context = context;
+};
+
+/**
+ * Listen for view events.
+ *
+ * @alias view#init
+ * @param {Object} opts - To set custom category, elementsToTrack, callback to collect data for the event
+ * @return {undefined}
+ */
+const init = (opts = {}) => {
+	if(!window.IntersectionObserver) {
+		return false;
+	}
+
+	const elementsToTrack = opts.elementsToTrack || [...document.querySelectorAll('[data-o-tracking-view]')];
+
+	function onChange (changes) {
+		changes.forEach(change => {
+			if(change.isIntersecting || change.intersectionRatio > 0) {
+				const eventData = {
+					action: 'view',
```

Comment:
I know we sort of settled on this now but I'm wondering if `component:seen` might be clearer........ what do you think? No worries if we're too far down the line!

Reply:
@benbarnett The business analyst side wants to keep consistent for the action.
`view` is already used for impressions. e.g. comment:view.